lib/Sema/CSDiagnostics.cpp

```diff
@@ -5573,6 +5573,15 @@ SourceLoc ExtraneousArgumentsFailure::getLoc() const {
   return FailureDiagnostic::getLoc();
 }
 
+ASTNode ExtraneousArgumentsFailure::getAnchor() const {
+  auto anchor = FailureDiagnostic::getAnchor();
+
+  if (auto *CE = getAsExpr<CoerceExpr>(anchor))
+    return CE->getSubExpr();
```

Comment:
I kind of wonder whether `CoerceExpr` should generate its conversion constraint with a locator element that points into the sub-expression, that way `simplifyLocator` would automatically handle this. That being said, I'm not sure what other impact that might have, so I'm happy to leave that for future work.